main.py

```diff
@@ -181,7 +196,9 @@ def run(self):
                 break
         g.init()
         pg.font.init()
-        self.heading = pg.font.Font(None, 96).render("Tic - Tac - Toe", True, g.WHITE)
+        self.heading = pg.font.Font(None, 96).render(
+            "Tic - Tac - Toe", True, g.WHITE)
+        self.turn_text = pg.font.Font(None, 64).render("X Turn", True, g.WHITE)
```

Comment:
The can be replaced by the line above in `check_events`.

Reply:
Hi Sparsh, initially this is what I did but the issue with that  approach would be the fact that self.frame isn't initialized within the scope of this block. and since player turn is stored in frame. it would lead to an error of us trying to access an undefined attribute